analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/FileMetadataAnalyzerTest.cs

```diff
@@ -126,27 +102,33 @@ public void CreateMessage_NoEncoding_SetsEmptyString(bool isTestProject)
             sut.TestCreateMessage(tree.Object, null).Encoding.Should().BeEmpty();
         }
 
-        private void VerifyAllFilesAreGenerated(string testName,
-                                                       string[] projectFiles,
-                                                       string[] autogeneratedFiles,
-                                                       ProjectType projectType,
-                                                       ImmutableArray<ParseOptions> parseOptions)
-        {
-            var testRoot = BasePath + testName;
+        [DataTestMethod]
+        [DataRow("class.generated.cs", 1)]  // ExtraEmptyFile.g.cs
+        [DataRow("SomethingElse.cs", 2)]    // ExtraEmptyFile.g.cs and class.generated.cs
```

Comment:
It took a while until I understood this test. I think we should provide 3 parameters instead:
- project files
- unchanged files
- count (without `ExtraEmptyFile.g.cs`, for it we can subtract 1 in the assert and add a comment there).

In this way, the full scenario is described on a single line.

If we use arrays we will be able to assert also scenarios with multiple files but it's not mandatory.

Reply:
I think it's better than before and, at the same time I think that something like:
```
        [DataRow("class.generated.cs", "class.generated.cs", 0)]
        [DataRow("class.generated.cs", "SomethingElse.cs", 1)]
        public void Verify_UnchangedFiles(string projectFileName, string unchangedFileName, int expectedFileCount) =>
```
would be even better in terms of clarity. 

We could even use arrays and then the test could assert scenarios with multiple files.

Reply:
The `ExtraEmptyFile.g.cs` is just a scaffolding noise that is ignored by the assertions. I wouldn't put more effort into it.